// services/item-service.js
function ItemService() {

}

ItemService.prototype = {

    constructor: ItemService,

    callRead: async function(id) {

        var self = this;

        let resultArray = await dao.read.call(id);
        return self.itemMapper(resultArray);
    },

    callReadByIndex: async function(index) {
        let resultArray = await dao.readByIndex.call(index);
        return self.itemMapper(resultArray);
    },


    callReadItemList: async function(limit, offset) {

        let currentCount = await dao.count();

        let items = [];

        for (var i=offset; (i < currentCount) || (i - offset == limit); i++) {
            let resultArray = await dao.readByIndex.call(i);
            items.push(itemMapper(resultArray));
        }

        return items;

    },

    itemMapper: function itemMapper(resultArray) {
        return {
            id: resultArray[0],
            owner: resultArray[1],
            version: resultArray[2],
            title: resultArray[3],
            inventory: resultArray[4],
            index: resultArray[5]
        }
    }

};

module.exports = ItemService;
